k8s/sonarqube/README.md

```diff
@@ -0,0 +1,503 @@
+```
+
+> **NOTE:** It is important to backup password.
+
+## Restoring
+
+Entering to maintenance mode:
+In order to restore PostgresSQL database there is a need to perform some preliminary steps:
+
+1. The command below will “lock” the database by blocking new incoming
+    connections:
+
+    ```shell
+    kubectl --namespace $NAMESPACE exec -t \
+      $(kubectl -n$NAMESPACE get pod -oname | \
+         sed -n /\\/$APP_INSTANCE_NAME-postgresql-deployment/s.pods\\?/..p) \
+      -c postgresql-server \
+      -- psql -U postgres -c "update pg_database set datallowconn = false where datname = 'sonar';"
+    ```
+
+1. Next in order to ensure data consistency all active connections will be dropped:
+
+    ```shell
+    kubectl --namespace $NAMESPACE exec -t \
+      $(kubectl -n$NAMESPACE get pod -oname | \
+         sed -n /\\/$APP_INSTANCE_NAME-postgresql-deployment/s.pods\\?/..p) \
+      -c postgresql-server \
+      -- psql -U postgres -c "select pg_terminate_backend(pid) from pg_stat_activity where datname='sonar';"
+    ```
+
+1. Now you are able to restore data to the database.
+Below shell script will restore data from `backup.sql` to PostgreSQL
+
+    ```shell
+    cat backup.sql | kubectl --namespace $NAMESPACE exec -i \
+      $(kubectl -n$NAMESPACE get pod -oname | \
+        sed -n /\\/$APP_INSTANCE_NAME-postgresql-deployment/s.pods\\?/..p) \
+      -c postgresql-server \
+      -- psql -U postgres
+    ```
+
+1. Next copy files from current folder to folder `$SONARQUBE_HOME/data` in SonarQube application pod:
+
+    ```shell
+    kubectl --namespace $NAMESPACE cp ./ $(kubectl -n$NAMESPACE get pod -oname | \
+      sed -n /\\/$APP_INSTANCE_NAME-sonarqube/s.pods\\?/..p):/opt/sonarqube/data
+    ```
+
+1. Afterwards copy files from current folder to `$SONARQUBE_HOME/extensions` folder in SonarQube application pod
+
+    ```shell
+    kubectl --namespace $NAMESPACE cp ./ $(kubectl -n$NAMESPACE get pod -oname | \
+      sed -n /\\/$APP_INSTANCE_NAME-sonarqube/s.pods\\?/..p):/opt/sonarqube/extensions
+    ```
+
+1. Delete all outstanding and unneeded SonarQube application data:
+
+    ```shell
+    kubectl --namespace $NAMESPACE exec -i \
+      $(kubectl -n$NAMESPACE get pod -oname | \
+        sed -n /\\/$APP_INSTANCE_NAME-sonarqube/s.pods\\?/..p) \
+      -- bash -c "rm -rf /opt/sonarqube/data/es5/* "
+    ```
+
+1. Exit maintenance mode by unlocking schema `sonar` for new incoming connection:
+
+    ```shell
+    kubectl --namespace $NAMESPACE exec -t \
+      $(kubectl -n$NAMESPACE get pod -oname | \
+        sed -n /\\/$APP_INSTANCE_NAME-postgresql-deployment/s.pods\\?/..p) \
+      -c postgresql-server \
+      -- psql -U postgres -c "update pg_database set datallowconn = true where datname = 'sonar';"
+    ```
+
+1. Patch secret to restore password:
+
+   ```shell
+   kubectl patch secret sonarqube-1-secret -p '{"data": {"password": "$ENCRYPTED_PASS"}}'
+   ```
+
+   where `$ENCRYPTED_PASS` value for secret.
```

Comment:
`$ENCRYPTED_PASS` is base64 encoded password, there is no encryption happening. Please change variable name.

Also, using `'` prevents `"$ENCRYPTED_PASS"` from evaluating - please provide a note about this, or change a line to `'{"data": {"password": "'"$ENCRYPTED_PASS"'"}}'`

Reply:
line changed